pkg/api/kubegraph/edges.go

```diff
@@ -0,0 +1,46 @@
+package kubegraph
+
+import (
+	"github.com/gonum/graph"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	kubegraph "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	deploygraph "github.com/openshift/origin/pkg/deploy/graph/nodes"
+)
+
+const (
+	ExposedThroughServiceEdgeKind = "ExposedThroughService"
+)
+
+// AddFullfillingDeploymentConfigEdges ensures that a directed edge exists between all deployment configs and the
+// services that expose them (via label selectors).
+func AddFullfillingDeploymentConfigEdges(g osgraph.MutableUniqueGraph, node *kubegraph.ServiceNode) *kubegraph.ServiceNode {
```

Comment:
> Why is this in kubegraph?

This is creating first generation edges that logically come from a `PodTemplateSpec`.  `DeploymentConfig` is due to bad containment that will be fixed.

Sort of related, does this edge seem backwards to you?